library/src/CMakeLists.txt

```diff
@@ -82,6 +82,11 @@ if(HIP_PLATFORM STREQUAL amd)
     endif( )
   endif( )
 
+  if(NOT BUILD_WITH_SOLVER)
+    target_link_libraries(hipblas PRIVATE $<$<PLATFORM_ID:Linux>:${CMAKE_DL_LIBS}>)
+    set_property(TARGET hipblas APPEND PROPERTY BUILD_RPATH ${ROCSOLVER_PATH}/lib)
```

Comment:
What prompted this change?

Reply:
I believe this is needed to find the rocsolver library if hipblas is built with the install script --rocsolver-path option

Reply:
Keep in mind that rocsolver might not actually even exist on the system at build-time. Personally, I wouldn't bother mucking around with the `RPATH`, as it can always be handled after the build by setting the `LD_LIBRARY_PATH`.

If you prefer to set the RPATH, it should be guarded to only be set when rocsolver exists and `--rocsolver-path` has been used. The hardcoded `lib` may be a problem, as the library directory can sometimes have other names in third-party builds, like `lib64` or `<target-triple>/lib`. You could use `${CMAKE_INSTALL_LIBDIR}` instead, but I think it would be more robust to extract the path from the target:
```cmake
if(DEFINED ROCSOLVER_PATH)
  find_package(rocsolver QUIET PATHS "${ROCSOLVER_PATH}" NO_DEFAULT_PATH)
  if(rocsolver_FOUND)
    get_target_property(rocsolver_location roc::rocsolver LOCATION)
    get_filename_component(rocsolver_directory ${rocsolver_location} DIRECTORY)
    set_property(TARGET hipblas APPEND PROPERTY BUILD_RPATH ${rocsolver_directory})
  endif()
endif()
```

With all that said, I prefer less code to more code. IMO, it's fine to expect users to set `LD_LIBRARY_PATH` if rocSOLVER is not in a system directory or `/opt/rocm`.